NewsApp/app/src/main/java/com/example/newsapp/features/auth/AuthViewModel.kt

```diff
@@ -1,8 +1,46 @@
 package com.example.newsapp.features.auth
 
-import androidx.lifecycle.ViewModel
+import androidx.lifecycle.*
+import com.example.newsapp.data.entities.User
 import com.example.newsapp.data.repos.UserRepository
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
 
 class AuthViewModel(
-    private val userRepository: UserRepository
-) : ViewModel()
\ No newline at end of file
+    private val userRepository: UserRepository,
+    private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO
+) : ViewModel() {
+    private val registrationResultMutableLiveData = MutableLiveData<Boolean>()
+    val registrationResultLiveData: LiveData<Boolean>
+        get() = registrationResultMutableLiveData
+
+    private val loginResultMutableLiveData = MutableLiveData<Boolean>()
+    val loginResultLiveData: LiveData<Boolean>
+        get() = loginResultMutableLiveData
+
+    private val changePasswordMutableLiveData = MutableLiveData<Boolean>()
```

Comment:
`changePasswordResultMutableLiveData` ?

Reply:
This was the original name but it was too long. But I will revert it to be consistent.